### PR TITLE
Ensure all goext resources inherit IResourceBase

### PR DESCRIPTION
**Pull request by przemyslaw-dobrowolski-cl**
This change ensures (during compilation) that RegisterType receives an object of type ResourceBase

### CODE REVIEW

File: extension/goext/schemas.go [hunk 1]
```diff
@@ -162,8 +162,8 @@ type ISchema interface {
 	// RegisterEventHandler registers an event handler for a named event with given priority
 	RegisterEventHandler(event string, handler func(context Context, resource Resource, environment IEnvironment) error, priority int)
 
-	// RegisterType registers a resource type, derived from BaseResource
-	RegisterType(resourceType interface{})
+	// RegisterType registers a resource type, derived from IResourceBase
+	RegisterType(resourceType IResourceBase)
 
 	// RegisterRawType registers a raw resource type, containing db annotations
 	RegisterRawType(rawResourceType interface{})
```
Comment (p-kozlowski) [inline extension/goext/schemas.go]: perhaps this is not the perfect moment, but we should keep in mind, that similar issues may happen in `RegisterRawType`, `(Db)[Create|Update](Raw)` and return types of `(Lock)[Fetch|List](Raw)`

File: converter/item/dbkind_test.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ var _ = Describe("db kind tests", func() {
 
 			result := dbKind.Type("", newItem)
 
-			expected := "goext." + util.ToGoName("null", typeOfItem)
+			expected := "goext." + util.ToGoName("maybe", typeOfItem)
```
Comment (morrisson) [inline converter/item/dbkind_test.go]: Duplicated #549 and should not be included in this PR.

**Comment by morrisson** [comment]
Please modify commit message and PR title. This does not describe general issue in open source context.

File: extension/goext/schemas.go (hunk 1 above)
Comment (przemyslaw-dobrowolski-cl) [reply extension/goext/schemas.go]: Good idea - I'll create a separate PR for that.

**Comment by przemyslaw-dobrowolski-cl** [comment]
@morrisson I fixed the description and removed unrelated commit from PR.
